docs/modules/ROOT/pages/explanations/decisions/user-management.adoc

```diff
@@ -0,0 +1,200 @@
+= Simple User management
+
+== Problem
+
+We want to provide easy user management for the AppCat services.
+The customer should be able to create users, permissions and if available for the given service, databases.
+The structure of the API should be similar for each service, just like the rest of the AppCat API is similar between services.
+
+=== Goals
+
+* The user management API should adhere to a convention spanning all services so that it follows the same structure for every service
+* Find a decision for simple service. There might be services in the future that have complex user and RBAC mechanisms that may not fit in this concept and need a custom decision
+* The API should be identical for the same service from different providers, given feature parity of the providers
+* Don't abstract service specific concepts in the API, this could lead to confusion
+
+=== Non-Goals
+
+* Having the exact same API for all services, it's unrealistic and too abstract
+
+== Proposals
+
+=== Generally
+Points outlined in this section apply to all proposals.
+
+When choosing a mechanism to manage the users, then we should prefer platform and provider-agnostic solutions before specific ones.
+For example `provider-sql` can manage PostgreSQL, MySQL and MSSQL directly.
+It only needs credentials to connect to the instance.
+In contrast, passing the user management to StackGres and let it handle the provisioning, would only work for PostgreSQL by VSHN, and we'd still have to come up with a solution for other PostgreSQL providers.
+
+If there are no crossplane providers or other operators to manage a service, but terraform modules are available, we can leverage Crossplane's https://github.com/crossplane/upjet[UpJet].
+
+When specifying users, then no passwords can be specified, only the names of the users.
```

Comment:
How do you intend to handle username collisions? On a shared DB instance, that might happen very soon. Maybe also generate usernames?

Reply:
As stated above, this is not a decision for shared systems.